gulpfile.js/import/importRoadmap.js

```diff
@@ -0,0 +1,116 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.ä
+ */
+
+'use strict';
+
+require('module-alias/register');
+
+const fs = require('fs');
+const yaml = require('js-yaml');
+const emojiStrip = require('emoji-strip');
+const {promisify} = require('util');
+const writeFileAsync = promisify(fs.writeFile);
+const {GitHubImporter} = require('@lib/pipeline/gitHubImporter');
+const log = require('@lib/utils/log')('Import Working Groups');
+
+/* The GitHub organisation the repositories imported from are located */
+const WG_GH_ORGANISATION = 'ampproject';
+/* Path where the roadmap data gets imported to */
+const ROADMAP_DIRECTORY_PATH = 'pages/shared/data';
+
+const ALLOWED_ISSUE_TYPES = ['Type: Status Update'];
+
+// RegEx to extract date from issue title
+const STATUS_UPDATE_REGEX = /(\d*)-(\d*)-(\d*)/;
+
+
+/**
+ * Extract status update issues from working groups and
+ * return them sorted by date/quarter
+ *
+ * @return []
+ */
+
+async function importRoadmap() {
+  var roadmap = [];
+
+  log.start('Importing Roadmap data for ..');
+
+  const client = new GitHubImporter();
+  const repos = (
+    await client._github.org(WG_GH_ORGANISATION).reposAsync(1, 100)
+  )[0];
+
+
+  // Get status update issues for working groups
+  for (const wg of repos) {
+    log.info('..', wg.name);
+
+    if (!wg.name.startsWith('wg-')) {
+      continue;
+    }
+    const workingGroupName = wg.name.substr(3);
+
+    let issues = (
+      await client._github.repo(`${WG_GH_ORGANISATION}/${wg.name}`).issuesAsync()
+    )[0];
+
+    issues = issues
+      .filter((issue) => {
+        const category = issue.labels[0] ? issue.labels[0].name : '';
+
+        return ALLOWED_ISSUE_TYPES.includes(category);
+      })
+      .map((issue) => {
+        const createdAt = new Date(issue.created_at).toDateString();
+        const title = emojiStrip(issue.title);
+
+        // Parse status update date from from issue title and set quarter
+        var statusUpdate = title.match(STATUS_UPDATE_REGEX);
+        let quarter;
+        if (statusUpdate) {
+          quarter = `Q:${Math.ceil(parseInt(statusUpdate[2]) / 3)} ${statusUpdate[1]}`;
+          statusUpdate = new Date(statusUpdate[0]).toDateString();
+        }
+
+        return {
+          'wg_name': workingGroupName,
+          'created_at': createdAt,
+          'status_update': statusUpdate || '',
+          'quarter': quarter || '',
+          'title': title,
+          'number': issue.number,
+          'author': issue.user.login,
+          'html_url': issue.html_url,
+          // 'body': issue.body,
+        };
+      });
+
+    roadmap.push(issues);
+  }
+
+  // Sort issues by date
+  roadmap = roadmap.flat().sort((a, b) => {
```

Comment:
Why is `flat` needed here?

Reply:
Because we don't want our status update issues grouped into sub-arrays by their working groups - this makes sorting easier and will reduce the number of iterations needed in frontend.